File: app/src/main/java/com/nfcalarmclock/statistics/NacStatisticTypeConverters.java
package com.nfcalarmclock.statistics;

import androidx.room.TypeConverter;
import java.util.Date;

/**
 * Type converters for when an object is retrieved from the database.
 */
public class NacStatisticTypeConverters
{

	/**
	 * Convert from a Long to a Date.
	 */
	@TypeConverter
	public static Date fromTimestamp(Long value)
	{
		return value == null ? null : new Date(value);
	}

	/**
	 * Convert from a Date to a Long.
	 */
	@TypeConverter
	public static Long dateToTimestamp(Date date)
	{
		return date == null ? null : date.getTime();
	}

}

File: app/src/main/kotlin/com/nfcalarmclock/statistics/db/NacStatisticTypeConverters.kt
package com.nfcalarmclock.statistics.db

import androidx.room.TypeConverter
import java.util.Date

/**
 * Type converters for when an object is retrieved from the database.
 */
object NacStatisticTypeConverters
{

	/**
	 * Convert from a Long to a Date.
	 */
	@TypeConverter
	fun fromTimestamp(value: Long): Date
	{
		return Date(value)
	}

	/**
	 * Convert from a Date to a Long.
	 */
	@TypeConverter
	fun dateToTimestamp(date: Date): Long
	{
		return date.time
	}

}